src/character/mod.rs

```diff
@@ -8,8 +8,7 @@ pub mod class;
 use crate::randomizer::{random, Randomizer};
 use class::Class;
 
-
-#[derive(Serialize, Deserialize, Debug, PartialEq)]
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone, Copy)]
```

Comment:
PartialEq trait because of tests. And Clone, Copy for a copied return from the `fn attack`.

Two doubts about this `enum Condition`:
- Should it be somewhere else? Have its own mod?
- Should I return a reference from the `fn attack`, associated to the lifetime of the receiver and remove Clone, Copy traits?

Reply:
I think it's ok to have it here for now. 
Regarding the other question I'd have to see the error to see if clone is really necessary, but in general I would stay away from lifetimes as much as possible